src/coreclr/vm/methodtable.inl

```diff
@@ -1342,12 +1359,14 @@ FORCEINLINE OBJECTREF MethodTable::GetManagedClassObjectIfExists()
     // Logging will be done by the slow path
     LOADERHANDLE handle = GetWriteableData_NoLogging()->GetExposedClassObjectHandle();
 
-    OBJECTREF retVal;
+    // First, check if have a cached reference to an effectively pinned (allocated on FOH) object
+    OBJECTREF retVal = GetPinnedManagedClassObjectIfExists();
```

Comment:
`GetManagedClassObjectIfExists` if perf critical method. It is used by `JIT_GetRuntimeType` that is still going to be used by unloadable and shared generic code. 

You want to make sure that it has same or better codegen compared to what it is today. I think it is significantly worse in the current state (`CanUnload` called by `GetPinnedManagedClassObjectIfExists` is virtual method call).

Reply:
Ah, good point, turns out the idea with the lowest bit is doable - switched to that